hawkmoth/parser.py

```diff
@@ -271,9 +286,18 @@ def _type_fixup(cursor):
         ttype = ''
         name = f'''{pad(ret_type)}({pad(stars_and_quals)}{cursor.spelling}{dims})({', '.join(args)})'''  # noqa: E501
     else:
-        ttype = cursor_type.spelling
+        type_elem = []
+
+        storage_class = _get_storage_class(cursor)
+        if storage_class:
+            type_elem.append(storage_class)
+
+        type_elem.append(cursor_type.spelling)
+
         if stars_and_quals:
-            ttype += ' ' + stars_and_quals
+            type_elem.append(stars_and_quals)
+
+        ttype = ' '.join(type_elem)
```

Comment:
The thing that bugs me about this is that the documentation will be different depending on whether the documentation comments are above declarations or definitions. Declarations will have extern, definitions won't.

I think it's a matter of style whether one wants to put the documentation in the headers (declarations) or in the implementation (definitions). I don't think that should impact the resulting documentation.

I'd suggest not including `extern` here at all.

I think `static` is useful here, though.

Reply:
> The thing that bugs me about this is that the documentation will be different depending on whether the documentation comments are above declarations or definitions. Declarations will have extern, definitions won't.
> 
> I think it's a matter of style whether one wants to put the documentation in the headers (declarations) or in the implementation (definitions). I don't think that should impact the resulting documentation.

Not just style. A user may want 2 sets of documentation: one for implementation details, one for the API users. Generally, I actually like having the distinction, though it is a difference from what we had been doing so far.

> I'd suggest not including `extern` here at all.

Hmm, if a header has an `extern` variable declared, that is API relevant I think. Of course, defining variables in a header is (almost) never a good idea so one could say the `extern` is redundant. But then there's that one case in which one needs to do just that. Not to mention I've seen a couple of cases where including `.c` files was the right thing to do...

> I think `static` is useful here, though.